info.go

```diff
@@ -127,6 +130,28 @@ func (pi *ProgramInfo) ID() (ProgramID, bool) {
 	return pi.id, pi.id > 0
 }
 
+// ProgramStats holds statistics of a program.
+type ProgramStats struct {
+	// Total accumulated runtime of the program ins ns.
+	RuntimeNS uint64
+	// Total number of times the program was called.
+	RunCount uint64
+}
+
+// Stats returns measured statistics of a program.
+//
+// Values can be 0 if collecting of statistics is not enabled.
+func (pi *ProgramInfo) Stats() (ProgramStats, error) {
+	info, err := bpfGetProgInfoByFD(pi.fd)
+	if err != nil {
+		return ProgramStats{}, err
+	}
+	return ProgramStats{
```

Comment:
Why not create this when doing the bpfGetProgInfoByFD here? https://github.com/cilium/ebpf/blob/7a313e22540feeda41df7b82432ab06c9ee50b78/info.go#L84
Instead of adding `*internal.FD` to `ProgramInfo` add `stats *ProgramStats`. Keeping the fd around like that without a `Close()` function is basically a leak.

This function can then be `Stats() (ProgramStats, bool)` with the following body:

```
if pi.stats != nil { return *pi.stats, true }
return ProgramStats{}, false
```

* A single syscall
* It's clear that ProgramStats may be absent (like on older kernels)
* No dangling fds

Reply:
Something I don't have a strong opinion on, but that is worth thinking about: is the `ProgramStats` type necessary? We could  have two getters on `ProgramInfo` instead: `ProgramInfo.Runtime() (foo, bool)` and `ProgramInfo.RunCount(foo, bool)`.

Reply:
Ok - I think I get your point.

The idea I had in mind was to be able to repeatly call `Stats()` and get the current information directly. While I understand your approach more like get `ProgramInfo` and from this get the statistics. With your approch getters are more usefull, I think.

Reply:
I've chosen `ProgramInfo.RunCount()(uint64)` over `ProgramInfo.RunCount()(uint64, bool)`, as we can not decide from the available data, if stats are enabled or not. e.g. `0` is a valid return if the program was never called.
If one can not call `EnableStats()` without error, then the kernel doesn't provide this information. So it is more an indirect check via the enable function.